aws_s3_bucket.tf

```diff
@@ -61,31 +73,47 @@ resource "aws_s3_bucket" "s3VersioningMfaFalse" {
       }
     }
   }
-  
+
   versioning {
-    enabled = false
-    mfa_delete = false
+    enabled    = true
+    mfa_delete = true
   }
 }
 
 resource "aws_s3_bucket" "allUsersReadAccess" {
   bucket = "my-tf-test-bucket"
-  acl    = "public-read"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
 }
 
 resource "aws_s3_bucket" "authUsersReadAccess" {
   bucket = "my-tf-test-bucket"
-  acl    = "authenticated-read"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
 }
 
 resource "aws_s3_bucket" "allUsersWriteAccess" {
   bucket = "my-tf-test-bucket"
-  acl    = "public-read-write"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
```

Comment:
**Accurics remediation**
 
**[HIGH]: S3 bucket should not be world-listable**
 
**Vulnerability**
Misconfigured S3 buckets can leak private information to the entire internet or allow unauthorized data tampering / deletion
 
**Remediation**
In the S3 console, select the Permissions tab, and then Bucket Policy. Remove policies for s3:List* actions for any principals '*'. If necessary, modify the policy instead, to limit the access to specific principals. Then, select Access Control List, and remove any grants to List objects to Public (everyone) users. Refer to: https://docs.aws.amazon.com/AmazonS3/latest/dev/using-iam-policies.html

source: accurics-